docs/source/spec-workload-executor.rst

```diff
@@ -37,55 +37,133 @@ Behavioral Description
+   
+   If the unified test runner reports a failure while executing the workload,
+   the failure MUST be reported using the same format as failures handled by the
+   unified test runner, as described in the unified test runner specification
+   under the ``loop`` operation. Failures handled by the workload
+   executor MUST be included in the calculated (and reported) failure count.
+   If the driver's unified test runner is intended to handle all failures
+   internally, failures that propagate out of the unified test runner MAY
+   be treated as errors by the workload executor.
+
+#. Upon receipt of the termination signal, MUST instruct the
+   unified test runner to stop looping, as defined in the unified test format.
+
+#. MUST wait for the unified test runner to finish executing.
+   
+#. MUST use the unified test runner to retrieve the following
+   entities by name from the entity map, if they are set:
+   
+   * ``iterations``: the number of iterations that the workload executor
+     performed over the looped operations. If the iteration count was not
+     reported by the test runner, such as because the respective option was
+     not specified in the test scenario, the workload executor MUST use
+     ``-1`` as the number of iterations.
+   
+   * ``successes``: the number of successful operations that the workload
+     executor performed over the looped operations. If the iteration count
+     was not reported by the test runner, such as because the respective
+     option was not specified in the test scenario, the workload executor
+     MUST use ``-1`` as the number of successes.
+   
+   * ``errors``: array of documents describing the errors that occurred
+     while the workload executor was executing the operations.
+   
+   * ``failures``: array of documents describing the failures that occurred
+     while the workload executor was executing the operations.
+   
+   * ``events``: array of documents describing the CMAP events that occurred
+     while the workload executor was executing the operations.
+
+   If the driver's unified test format does not distinguish between errors
+   and failures, and reports one but not the other, the workload executor MUST
+   set the non-reported entry to the empty array.
+
+#. MUST calculate the aggregate counts of errors (``numErrors``) and failures
+   (``numFailures``) from the error and failure lists. If the errors or
+   failures were not reported by the test runner, such as because the
+   respective options were not specified in the test scenario, the workload
+   executor MUST use ``-1`` as the value for the respective counts.
+
+#. MUST write the collected events, errors and failures into a JSON file named
+   ``events.json`` in the current directory
+   (i.e. the directory from where the workload executor is being executed). 
+   The data written MUST be a map with the following fields:
+   
+   - ``events``: the collected CMAP events.
```

Comment:
The [Design Update](https://docs.google.com/document/d/1nvIY5byfFdvIYiC1XLz964U0NlvkkX_OEv1dFMronJ4/edit) was revised to refer to command monitoring events, but this doesn't talk about those events at all. Was correlation of command events (e.g. calculating durations) supposed to be added to the workload executor's responsibilities in this PR, or was that pushed to some follow-up ticket?

Reply:
Correlation is performed by astrolabe, workload executor only needs to persist events as they are produced by the test runner.